software/deploy/deploy.go

```diff
@@ -296,6 +240,74 @@ func getAirflowUILink(deploymentID string, deploymentURLs []houston.DeploymentUR
 	return ""
 }
 
+func getDeploymentIDForCurrentCommand(houstonClient houston.ClientInterface, wsID, deploymentID string, prompt bool) (string, []houston.Deployment, error) {
+	if wsID == "" {
+		return deploymentID, []houston.Deployment{}, ErrNoWorkspaceID
+	}
+
+	// Validate workspace
+	currentWorkspace, err := houston.Call(houstonClient.GetWorkspace)(wsID)
+	if err != nil {
+		return deploymentID, []houston.Deployment{}, err
+	}
+
+	// Get Deployments from workspace ID
+	request := houston.ListDeploymentsRequest{
+		WorkspaceID: currentWorkspace.ID,
+	}
+	deployments, err := houston.Call(houstonClient.ListDeployments)(request)
+	if err != nil {
+		return deploymentID, deployments, err
+	}
+
+	c, err := config.GetCurrentContext()
+	if err != nil {
+		return deploymentID, deployments, err
+	}
+
+	cloudDomain := c.Domain
+	if cloudDomain == "" {
+		return deploymentID, deployments, errNoDomainSet
```

Comment:
Tricky to cover this and below lines. Since these lines are already incovered and I just refactored this part, I am not covering these lines in this PR. @kushalmalani @vandyliu